src/edu/harvard/CS50.java

```diff
@@ -39,12 +39,41 @@
 
 package edu.harvard;
 
+import java.io.BufferedOutputStream;
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.util.Scanner;
 
 public class CS50
 {
+    // disable buffering on stderr and stdout
+    // only functional when class is initialized per http://docs.oracle.com/javase/specs/jls/se7/html/jls-12.html#jls-12.4.1
+    static
```

Comment:
Well, per the docs, `FileOutputStream(FileDescriptor fdObj)` may throw `NULLPointerException` or `SecurityException` in case `fdObj` is `null` or [`checkWrite`](https://docs.oracle.com/javase/7/docs/api/java/lang/SecurityManager.html#checkWrite(java.io.FileDescriptor)) denies write access to the descriptor.

The descriptors here are for [stderr](https://docs.oracle.com/javase/7/docs/api/java/io/FileDescriptor.html#err) and [stdout](https://docs.oracle.com/javase/7/docs/api/java/io/FileDescriptor.html#out) and the docs never mentions any situations in which these can be `null`.

Furthermore, if stderr and/or stdout are non-writable initially, this should be caught before the JVM is started? If feels like good measure to handle, I can add a `try/catch`, but it's a deal-breaker if stdout ends up non-writable after all?

`BufferedOutputStream`'s [constructor](https://docs.oracle.com/javase/7/docs/api/java/io/BufferedOutputStream.html#BufferedOutputStream(java.io.OutputStream,%20int)) throws an `IllegalArgumentException` if size is <= 0, which we don't need to worry about, since we're hard-coding 1. And `PrintStream`'s [constructor](https://docs.oracle.com/javase/7/docs/api/java/io/PrintStream.html#PrintStream(java.io.OutputStream)) doesn't thrown any exceptions.